sources/packages/backend/libs/test-utils/src/factories/disbursement-receipt-value.ts

```diff
@@ -0,0 +1,23 @@
+import { DisbursementReceipt, DisbursementReceiptValue } from "@sims/sims-db";
+
+/**
+ * Create fake disbursement receipt value.
+ * @param grantType type of grant to be created.
+ * @param grantAmount grant amount to be created.
+ * @param relations dependencies.
+ * - `disbursementReceipt` related disbursement receipt.
+ * @returns disbursement receipt value.
+ */
+export function createFakeDisbursementReceiptValue(
+  grantType: string,
```

Comment:
As the team agreed we would not be exposing parameters for the sole purpose of directly setting properties.

Reply:
I have seen this again and again in the PR. Maybe we can try an alternative approach to avoid having all the properties exposed and still allow the initialization.
```ts
export function createFakeDisbursementReceiptValue(
  initialValues: Partial<DisbursementReceiptValue>,
  relations: {
    disbursementReceipt: DisbursementReceipt;
  },
): DisbursementReceiptValue {
  const disbursementReceiptValue = new DisbursementReceiptValue();
  disbursementReceiptValue.grantType = initialValues.grantType;
  disbursementReceiptValue.grantAmount = initialValues.grantAmount;
  disbursementReceiptValue.disbursementReceipt = relations.disbursementReceipt;
  return disbursementReceiptValue;
}
```
and then the consumer would have access to set all the properties.
```ts
createFakeDisbursementReceiptValue(
  {
    grantType: disbursement.valueCode,
    grantAmount: disbursement.valueAmount,
  },
  { disbursementReceipt },
)
```

Reply:
how about adding side options?
![image](https://github.com/bcgov/SIMS/assets/77353155/ac94c079-0dad-4ec4-a197-cc61d47272e7)

Reply:
Moving it to options will possibly move the issue to another part and for every single option, we will also need to add more comments.
The possible issue is just about a potential entity having too many properties exposed as parameters.
Wonder if we expose some parameters for `createFakeDisbursementReceipt` and later another developer exposes a few more parameters, and later another developer exposes a few more, and so on. It could become:
```ts
createFakeDisbursementReceipt("something", 1, true, false, "someStatue", "The Other Status", 1000, 456, 789, false);
```

Reply:
I am okay, we can handle them in the calling method by passing the disbursementReceipt @andrewsignori-aot as you said, rather passing every other parameter. In that way, we can restrict developers when enhancing the createFakeDisbursementReceipt, the signature will not be changed.

Reply:
team, pls share your thoughts

Reply:
I also think that using initialValues `initialValues: Partial<DisbursementReceiptValue>` will make it easier to send initial values to factory. 

@ann-aot I did a similar review adjustment in my last PR because I was just using the options to assign a property directly to the object.